packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3731,7 +3731,17 @@ class StatefulElement extends ComponentElement {
     assert(_state._debugLifecycleState == _StateLifecycle.created);
     try {
       _debugSetAllowIgnoredCallsToMarkNeedsBuild(true);
-      _state.initState();
+      final void _debugCheckForReturnedFuture = _state.initState();
+      assert(() {
+        if ((_debugCheckForReturnedFuture  as dynamic) is Future) {
+          throw new FlutterError(
+              '${_state.runtimeType}.didUpdateWidget returned a Future.\n'
```

Comment:
I don't think we use 4-space indent pretty much anywhere. It's 2 spaces, and in the situations where you would use 4 spaces, we prefer either breaking the 80-character limit, or split the expression into smaller expressions. We do like horizontal alignment though, e.g. in initializer lists:

```dart
class Foo {

  Foo({
    this.bar,
    this.baz,
  }): assert(bar != null),
      assert(baz != null);
//    ^
//    happens to be 4 characters but only because it aligns with the previous line
}
```

Reply:
Fixed